modules/gloo/pkg/vsfactory/target.go

```diff
@@ -51,32 +55,32 @@ func getTarget(obj *riov1.Service, systemNamespace string) (result target, err e
 		}
 	}
 
-	if obj.Status.ComputedReplicas != nil && *obj.Status.ComputedReplicas == 0 {
-		result.ScaleIsZero = true
-		result.Port = 80
-		result.OriginalTarget = struct {
-			Name      string
-			Namespace string
-		}{Name: obj.Name, Namespace: result.Namespace}
-		result.Name = constants.AutoscalerServiceName
-		result.Namespace = systemNamespace
+	result.Hosts, err = determineHosts(obj.Status.Endpoints)
+	if err != nil {
+		logrus.Warnf("%v", err)
+		return result, err
 	}
 
-	seen := map[string]bool{}
-	for _, endpoint := range obj.Status.Endpoints {
-		u, err := url.Parse(endpoint)
+	if obj.Status.ComputedReplicas != nil { // valid candidate for a service that can be scaled to zero
+		result.ScaleIsZero, err = f.isScaleZero(app, version, result.Namespace)
 		if err != nil {
 			return result, err
 		}
-		if seen[u.Host] {
-			continue
+		if result.ScaleIsZero {
+			if *obj.Status.ComputedReplicas > 0 {
+				logrus.Debug("service has ComputedReplicas > 0 but no IP address allocated via k8s endpoints yet")
+			}
+			result.Port = 80
```

Comment:
So basically this whole method first checks if ComputedReplicas is >= 0, and then checks if there are any endpoints. Then if there aren't endpoints it returns a target, thats the actual change here. Previously it only cared about reps. That right ?

Reply:
Now we wait for IPs _and_ computedReplicas of >= 0. Not nil computedReplicas because we want it to have been scaled at least once and then downscaled in order to get the IP ? Is there a reason we need both checks ? 

`isScaleZero` feels unintuitive since its not really dealing with scale specifically, its dealing with the "zero-scale" state which affects other fields/resources. Wish we had a separate word for that. Could rename it to `noEndpoints`, or `noIP`, or just `isZeroScaled`, but I guess its fine as-is also.

Reply:
nil `computedReplicas` is an error state since nil `computedReplicas` indicates that the service is not a canidate for autoscaling. If you have nil `computedReplicas` and a scale of zero there is an issue outside of this router code.

Reply:
> Now we wait for IPs and computedReplicas of >= 0
If computeReplics >= 0 we only print a debug log. No impact on logic